index.js

```diff
@@ -1,5 +1,67 @@
 'use strict';
 
+var superagent = require('superagent');
+
+global.D2LAccessTokenExpiresAt = 0;
+
+function now() {
+	return Date.now()/1000 | 0;
+}
+
+function addHeaders(req) {
+	req.set('X-Csrf-Token', localStorage['XSRF.Token']);
+	req.set('X-D2L-App-Id', 'deprecated');
+}
+
+function processRefreshResponse(cb) { return function(res) {
+	if (!res.ok) {
+		return cb(false);
+	}
+
+	var cacheControl = res.headers['cache-control'];
+	if (!cacheControl) {
+		return cb(true);
+	}
+
+	var directives = cacheControl.split(',');
+	for (var i = 0; i < directives.length; i++) {
+		if (directives[i].indexOf('max-age') == -1) {
+			continue;
+		}
+
+		var maxAge = +directives[i].split('=')[1];
```

Comment:
This assumed `max-age` is not malformed an a number actually follows the `=` (probably true I guess).

Might be good to check for `!isNaN(maxAge)` though?